src/zod-factory.ts

```diff
@@ -0,0 +1,307 @@
+import type { LocaleDefinition, Randomizer } from '@faker-js/faker';
+import { en, Faker } from '@faker-js/faker';
+import { z } from 'zod';
+
+export interface ZodFactoryConfig {
+  /**
+   * Custom generators for specific schema descriptions
+   */
+  customGenerators?: Record<string, () => unknown>;
+  
+  /**
+   * Faker.js locale to use for generation
+   */
+  locale?: LocaleDefinition;
+  
+  /**
+   * Custom randomizer for Faker.js
+   */
+  randomizer?: Randomizer;
+}
+
+type ZodSchema = z.ZodTypeAny;
+
+/**
+ * Creates a factory function from a Zod schema that generates type-safe mock data
+ */
+export function createFactoryFromZod<T extends ZodSchema>(
+  schema: T,
+  config: ZodFactoryConfig = {}
+): {
+  generate(): z.infer<T>;
+  batch(count: number): z.infer<T>[];
+  build(overrides?: Partial<z.infer<T>>): z.infer<T>;
+  buildMany(count: number): z.infer<T>[];
+} {
+  const faker = new Faker({
```

Comment:
you are wrapping faker, but our base factory class extends the Faker class directly. 

This is the approach we should build on. 

ZodFactory should extend our Factory class with new functionalities. 

You are in effect introducing a second paradigm into the codebase this way.